src/pelicanfs/core.py

```diff
@@ -298,36 +329,171 @@ def _dirlist_dec(func):
         Decorator function which, when given a namespace location, get the url for the dirlist location from the headers
         and uses that url for the given function.
 
-        This is for functions which need to list information in the origin directories such as "find", "isdir", "ls"
+        This is for functions which need to get information from origin directories "ls", "du", "info", etc.
         """
         async def wrapper(self, *args, **kwargs):
-            path = args[0]
-            path = self._check_fspath(path)
-            # TODO: need to have a separate get_dirlist_url
-            listUrl = await self.get_origin_url(path)
-            result = await func(self, listUrl, *args[1:], **kwargs)
+            path = self._check_fspath(args[0])
+            dataUrl = await self.get_dirlist_url(path)
+            result = await func(self, dataUrl, *args[1:], **kwargs)
             return result
         return wrapper
 
+
+    def _remove_dirlist_from_path(self, path):
+        parsed_url = urllib.parse.urlparse(path)
+        updated_url = parsed_url._replace(netloc="", scheme="")
+        return urllib.parse.urlunparse(updated_url)
+
+    def _remove_dirlist_from_paths(self, paths):
+        if isinstance(paths, list):
+            return [self._remove_dirlist_from_paths(path) for path in paths]
+
+        if isinstance(paths, dict):
```

Comment:
I'm not sure this is the best way to see if something is "dictionary-like" in Python3.  Do a quick sanity check with @matyasselmeci -- if there's not an obvious better way to do it, you can keep it here.